airbyte-cdk/python/airbyte_cdk/utils/traced_exception.py

```diff
@@ -0,0 +1,51 @@
+#
+# Copyright (c) 2021 Airbyte, Inc., all rights reserved.
+#
+
+import traceback
+from datetime import datetime
+
+from airbyte_cdk.models import AirbyteErrorTraceMessage, AirbyteMessage, AirbyteTraceMessage, FailureType, TraceType
+from airbyte_cdk.models import Type as MessageType
+
+
+class AirbyteTracedException(Exception):
+    """
+    TODO: give this a meaningful description
+    """
+
+    def __init__(
+        self,
+        internal_message: str = None,
+        message: str = None,
+        failure_type: FailureType = FailureType.system_error,
+        exception: BaseException = None,
+    ):
+        self.internal_message = internal_message
+        self.message = message
+        self.failure_type = failure_type
+        self._exception = exception
+        super().__init__(internal_message)
+
+    def as_airbyte_message(self) -> AirbyteMessage:
+        now_millis = int(datetime.now().timestamp() * 1000)
```

Comment:
this is a Double in the protocol so should be a `float` here in python

Reply:
Oh interesting - the `RecordMessage`'s `emitted_at` is an `int` 🤔  good catch!